presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -731,6 +732,45 @@ public static double chiSquaredCdf(
         return distribution.cumulativeProbability(value);
     }
 
+    @Description("Hypergeometric cdf given populationSize, numberOfSuccesses, sampleSize and value")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double hypergeometricCdf(
+            @SqlType(StandardTypes.INTEGER) long populationSize,
+            @SqlType(StandardTypes.INTEGER) long numberOfSuccesses,
+            @SqlType(StandardTypes.INTEGER) long sampleSize,
+            @SqlType(StandardTypes.INTEGER) long value)
+    {
+        checkCondition(populationSize > 0, INVALID_FUNCTION_ARGUMENT, "populationSize must be greater than 0");
+        checkCondition(numberOfSuccesses >= 0, INVALID_FUNCTION_ARGUMENT, "numberOfSuccesses cannot be negative");
+        checkCondition(numberOfSuccesses <= populationSize, INVALID_FUNCTION_ARGUMENT, "numberOfSuccesses cannot exceed populationSize");
```

Comment:
nits: Merge this so the error message is `numberOfSuccesses should be in range [0, populationSize]` maybe?